### PR TITLE
Add checks for mint and owner accounts

### PR DESCRIPTION
**Pull request by joncinque**
We don't check that the mint and owner accounts are actually mint and owner accounts, so let's add some checks and helpful errors!

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -177,13 +212,18 @@ fn main() {
...
-            let (owners, mints) = get_owners_and_mints(sub_matches, true, &mut wallet_manager);
+            let (owners, mints) =
+                get_owners_and_mints(sub_matches, true, &config.rpc_client, &mut wallet_manager);
             audit::run(config, owners, mints);
         }
         ("cleanup", Some(sub_matches)) => {
-            let allow_null_signer = if dry_run { true } else { false };
-            let (owners, mints) =
-                get_owners_and_mints(sub_matches, allow_null_signer, &mut wallet_manager);
+            let allow_null_signer = dry_run;
```
Comment (joncinque) [inline src/main.rs:223]: clippy complained about this

Comment (ryoqun) [reply src/main.rs:223]: heh, i did only `cargo fmt`... xD

Comment (joncinque) [reply src/main.rs:223]: haha sorry, if it were up to me, i wouldn't have changed it!

File: src/main.rs [hunk 2]
```diff
@@ -77,6 +98,20 @@ fn get_owners_and_mints(
         .unwrap()
         .map(|p| get_signer(sub_matches, p, wallet_manager, allow_null_signer))
         .collect::<Vec<_>>();
+
+    for owner in &owners {
+        let owner_address = owner.pubkey();
+        if let Ok(owner_account) = rpc_client.get_account(&owner_address) {
```
Comment (ryoqun) [inline src/main.rs:107]: nits: `// don't unwrap to allow possibly not-existing owner`?

File: src/main.rs [hunk 3]
```diff
@@ -50,24 +53,42 @@ fn get_signer(
...
+            let mint_account = rpc_client.get_account(mint).unwrap_or_else(|_|
+                panic!(
+                    "Account {} expected to be an SPL token mint, but does not exist. Maybe this is a system account?",
+                    mint
                     )
-                    .pubkey()
-                })
-                .collect::<Vec<_>>(),
-        )
+                );
```
Comment (ryoqun) [inline src/main.rs]: odd indent?

Comment (joncinque) [reply src/main.rs]: it was very strange, i changed this a few different ways, and cargo fmt did nothing with it... let me see if i can get it like other one

**Review by ryoqun — APPROVED** [review]
lgtm with nits

**Pull request merged**